src/Tree.zig

```diff
@@ -438,11 +440,20 @@ pub fn isLval(nodes: Node.List.Slice, node: NodeIndex) bool {
         },
         .member_access_expr => {
             const data = nodes.items(.data)[@enumToInt(node)];
-            return isLval(nodes, data.member.lhs);
+            return isLval(nodes, extra, value_map, data.member.lhs);
         },
         .paren_expr => {
             const data = nodes.items(.data)[@enumToInt(node)];
-            return isLval(nodes, data.un);
+            return isLval(nodes, extra, value_map, data.un);
+        },
+        .builtin_choose_expr => {
+            const data = nodes.items(.data)[@enumToInt(node)];
+
+            if (value_map.get(data.if3.cond)) |val| {
```

Comment:
This is a debugging fearure and should print bith branches unconditionally. The chosen branch can easily be seen by looking at the saved value of the condition.

Reply:
For the printer we could do that but isn't this function also used by the parser for deciding if things can be assigned to / have their address taken?

Reply:
My bad, thought this was `dump`. I really should double-check my comments when doing reviews in the morning.